api/controller.py

```diff
@@ -899,33 +900,44 @@ def download_page(self):
         if len(library.cachedmarcfiles) < 1 and exporter:
             body += "<p>" + _("MARC files aren't ready to download yet.") + "</p>"
 
-        full_files = self._db.query(CachedMARCFile).filter(
-            CachedMARCFile.library==library
-        ).filter(
-            CachedMARCFile.start_time==None
-        )
+        files_by_lane = defaultdict(dict)
+        for file in library.cachedmarcfiles:
+            if file.start_time == None:
+                files_by_lane[file.lane]["full"] = file
+            else:
+                if not files_by_lane[file.lane].get("updates"):
+                    files_by_lane[file.lane]["updates"] = []
+                files_by_lane[file.lane]["updates"].append(file)
```

Comment:
This makes sense to me if creating a new full MARC file wipes out the previous full file and the updates. Is that right?

Reply:
OK, I get it, thanks.